userland/libtock/ble.h

```diff
@@ -2,125 +2,140 @@
-    NEG_30_DBM   = 0xD8
-} BLE_TX_Power_t;
+  GAP_FLAGS                              = 0x01, /* Flags, see enum below */
+  GAP_INCOMPLETE_LIST_16BIT_SERVICE_IDS  = 0x02, /* Incomplete list of 16-bit Service IDs. */
+  GAP_COMPLETE_LIST_16BIT_SERVICE_IDS    = 0x03, /* Complete list of 16-bit Service IDs. */
+  GAP_INCOMPLETE_LIST_32BIT_SERVICE_IDS  = 0x04, /* Incomplete list of 32-bit Service IDs (not relevant for Bluetooth 4.0). */
+  GAP_COMPLETE_LIST_32BIT_SERVICE_IDS    = 0x05, /* Complete list of 32-bit Service IDs (not relevant for Bluetooth 4.0). */
+  GAP_INCOMPLETE_LIST_128BIT_SERVICE_IDS = 0x06, /* Incomplete list of 128-bit Service IDs. */
+  GAP_COMPLETE_LIST_128BIT_SERVICE_IDS   = 0x07, /* Complete list of 128-bit Service IDs. */
+  GAP_SHORTENED_LOCAL_NAME               = 0x08, /* Shortened Local Name. */
+  GAP_COMPLETE_LOCAL_NAME                = 0x09, /* Complete Local Name. */
+  GAP_TX_POWER_LEVEL                     = 0x0A, /* TX Power Level (in dBm). */
+  GAP_DEVICE_ID                          = 0x10, /* Device ID. */
+  GAP_SLAVE_CONNECTION_INTERVAL_RANGE    = 0x12, /* Slave Connection Interval Range. */
+  GAP_LIST_128BIT_SOLICITATION_IDS       = 0x15, /* List of 128 bit service UUIDs the device is looking for. */
+  GAP_SERVICE_DATA                       = 0x16, /* Service Data. */
+  GAP_APPEARANCE                         = 0x19, /* Appearance */
+  GAP_ADVERTISING_INTERVAL               = 0x1A, /* Advertising Interval. */
+  GAP_MANUFACTURER_SPECIFIC_DATA         = 0xFF  /* Manufacturer Specific Data. */
+} GapAdvertisementData_t;
+
+enum {
+  LE_LIMITED_DISCOVERABLE = 0x01, /* Peripheral device is discoverable for a limited period of time. */
+  LE_GENERAL_DISCOVERABLE = 0x02, /* Peripheral device is discoverable at any moment. */
+  BREDR_NOT_SUPPORTED     = 0x04, /* Peripheral device is LE only. */
+  SIMULTANEOUS_LE_BREDR_C = 0x08, /* Not relevant - central mode only. */
+  SIMULTANEOUS_LE_BREDR_H = 0x10  /*Not relevant - central mode only. */
+} GapFlags_t;
+
+
+
+
+/*******************************************************************************
+ *   User-space API
+ ******************************************************************************/
+
+ // initialize advertisement (should be used before invoked ble_start_advertising)
+ //
+ // advertising_iterval_ms  - advertisement interval in milliseconds
+ // discoverable            - if the device should be discoverable or not
+ // addr                    - address to be used in the advertisment
+ // addr_len                - length of the address in bytes (it should be 6)
+int ble_initialize(uint16_t advertising_interval_ms, bool discoverable,
+                  uint8_t *addr, uint8_t addr_len);
+
+
+// start advertising
+int ble_start_advertising(void);
+
+
+// stop advertising but don't change anything in the packet configuration
+int ble_stop_advertising(void);
+
+// reset the entire advertisement data packet
+// excluding flags and advertisement address
```

Comment:
Does this also implicitly stop advertising?

Reply:
No, but it perhaps should?